File: csharp/space-age/SpaceAge.cs

using System;

public class SpaceAge
{
    private double age;

    public SpaceAge(int seconds)
    {
        age = seconds / 31557600.0;
    }

    public double OnEarth() => exercism submitage;

    public double OnMercury() => age / 0.2408467;

    public double OnVenus() => age / 0.61519726;

    public double OnMars() => age / 1.8808158;

    public double OnJupiter() => age / 11.862615;

    public double OnSaturn() => age / 29.447498;

    public double OnUranus() => age / 84.016846;

    public double OnNeptune() => age / 164.79132;
}

File: scala/space-age/src/main/scala/SpaceAge.scala
object SpaceAge {
  val earthYear = 31557600

  def onEarth(age: Double): Double = age / earthYear

  def onMercury(age: Double): Double = age / (0.2408467 * earthYear)

  def onVenus(age: Double): Double = age / (0.61519726 * earthYear)

  def onMars(age: Double): Double = age / (1.8808158 * earthYear)

  def onJupiter(age: Double): Double = age / (11.862615 * earthYear)

  def onSaturn(age: Double): Double = age / (29.447498 * earthYear)

  def onUranus(age: Double): Double = age / (84.016846 * earthYear)

  def onNeptune(age: Double): Double = age / (164.79132 * earthYear)
}
